src/Compilers/CSharp/Portable/Binder/Binder_Patterns.cs

```diff
@@ -266,10 +266,13 @@ private BoundPattern BindDiscardPattern(DiscardPatternSyntax node, TypeSymbol in
                     diagnostics.Add(ErrorCode.ERR_ConstantExpected, patternExpression.Location);
                     hasErrors = true;
                 }
-                else if (inputType.IsPointerType() && Compilation.LanguageVersion < MessageID.IDS_FeatureRecursivePatterns.RequiredVersion())
+                else if (inputType.IsPointerType() && Compilation.LanguageVersion < MessageID.IDS_FeatureNullPointerConstantPattern.RequiredVersion())
                 {
                     // before C# 8 we did not permit `pointer is null`
-                    diagnostics.Add(ErrorCode.ERR_PointerTypeInPatternMatching, patternExpression.Location);
+                    diagnostics.Add(ErrorCode.ERR_FeatureNotAvailableInVersion7_3,
```

Comment:
The other thing you want to do is add this Feature to our existing list and, rather than explicitly creating a diagnostic, use the existing facilities that will create it for you if required. These are here: http://sourceroslyn.io/#Microsoft.CodeAnalysis.CSharp/Errors/MessageID.cs,3a72b9253a37b9f0,references